### ISSUE TITLE
Encountering an mdDoc related error when updating nix-darwin via flake

### ISSUE THREAD

**Issue by yoav-lavi** [LnL7/nix-darwin#933]
```
error:
       … while calling the 'derivationStrict' builtin

         at /builtin/derivation.nix:9:12: (source not available)

       … while evaluating derivation 'darwin-system-24.05.20240414.2748d22+darwin4.36524ad'
         whose name attribute is located at /nix/store/qjgi80svyhfcwbnlvzzaqg0nyr7h2w59-source/pkgs/stdenv/generic/make-derivation.nix:331:7

       … while evaluating attribute 'activationScript' of derivation 'darwin-system-24.05.20240414.2748d22+darwin4.36524ad'

         at /nix/store/87clbsxcf2wawlhi49k96ffb805sa3v0-source/modules/system/default.nix:95:7:

           94|
           95|       activationScript = cfg.activationScripts.script.text;
             |       ^
           96|       activationUserScript = cfg.activationScripts.userScript.text;

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: attribute 'mdDoc' missing

       at /nix/store/87clbsxcf2wawlhi49k96ffb805sa3v0-source/modules/documentation/default.nix:121:21:

          120|       default = true;
          121|       description = lib.mdDoc ''
             |                     ^
          122|         Whether to install documentation distributed in packages' `/share/doc`.
```

**Comment by purcell** [comment]
I see that in `home-manager` there has been [at least one recent change](https://github.com/nix-community/home-manager/pull/5265/files) _removing_ references to `lib.mdDoc`, following up on [bulk changes last July](https://github.com/nix-community/home-manager/pull/4215). Now `lib.mdDoc` has been removed from upstream Nixpkgs entirely: https://github.com/NixOS/nixpkgs/pull/303841

So the fix here should be to remove the `lib.mdDoc` falls, which should already be redundant. I'll make a quick PR.

**Comment by purcell** [comment]
Ah, in fact there's already #932.

**Comment by purcell** [comment]
You can update your flake source to point the URL for `darwin` at `"github:wegank/nix-darwin/mddoc-remove"` — I've done this myself and #932 indeed solves the problem for me.

**Comment by yonkeltron** [comment]
> You can update your flake source to point the URL for `darwin` at `"github:wegank/nix-darwin/mddoc-remove"` — I've done this myself and #932 indeed solves the problem for me.

Fantastic, testing it now on the clean install using the flake method. Feedback forthcoming. Please stand by!

**Comment by yonkeltron** [comment]
This works great, I have run a couple of little tests locally and everything seems perfecto. Thanks a ton for the effort!

Love,
+Jonathan

**Comment by purcell** [comment]
> Thanks a ton for the effort!
> ...
> Although I notice that pkg-config seems to be broken at first glance. 

I'm not the author of PR #932, so I'd suggest leaving any feedback there. 😁

**Comment by wegank** [comment]
The `lib.mdDoc` removal has been deferred to 24.11 in nixpkgs (NixOS/nixpkgs#304277), so the PR is likely unneeded for now.

**Comment by rkitover** [comment]
Unstable channel still broken.

**Comment by yonkeltron** [comment]
This appears to be fixed by https://github.com/LnL7/nix-darwin/pull/932 on my macOS device.

**Comment by Enzime** [comment]
Should be fixed by #932
